### PR TITLE
Update mixlib-install 2.0 for PackageRouter updates

### PR DESCRIPTION
**Pull request by wrightp**
Update mixlib-install to `~> 2.0`

Includes some test cleanup:
- update kitchen-ec2 tag name
- allow travis to run on all branches
- Remove Gemfile (chefdk!)
- Update chefdk integration test to handle upgrades

@chef-cookbooks/engineering-services

### CODE REVIEW

File: .kitchen.ec2.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ end
 driver:
   name: ec2
   tags:
-    X-Project: chef-ci-acceptance
+    X-Project: chef-ingredient-acceptance
```
Comment (yzl) [inline .kitchen.ec2.yml]: Do you need the X-Dept and X-Contact tags as well?  
https://chefio.atlassian.net/wiki/display/ENG/AWS+Tagging+Standard

Comment (wrightp) [inline .kitchen.ec2.yml]: Thank you. I will add those!

**Comment by rhass** [comment]
It seems we should bump the version in the metadata to 0.21.0 (or a major version for that matter) to reflect this change as well.

Other than that I am :+1:

**Comment by wrightp** [comment]
@rhass I like to separate feature PRs from release commits. Once this is merged I'll update the version and publish the cookbook. As for the version, we will be bumping to `0.21.0`.

**Pull request merged**